src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

```diff
@@ -25,4 +29,53 @@ public Film updateFilm(Film film) {
     public List<Film> getFilms() {
         return filmStorage.getFilms();
     }
+
+    public Film getFilmById(Long id) {
+        return filmStorage.getFilmById(id);
+    }
+
+    public void deleteFriend(Long userId, Long friendId) {
+        User user = userStorage.getUserById(userId);
+        User friend = userStorage.getUserById(friendId);
+        user.getFriends().remove(friendId);
+        friend.getFriends().remove(userId);
+    }
+
+    public List<User> getFriends(Long userId) {
+        User user = userStorage.getUserById(userId);
+        return user
+                .getFriends()
+                .stream()
+                .map(userStorage::getUserById)
+                .collect(Collectors.toList());
+    }
+
+    public List<User> getCommonFriends(Long userId, Long otherUserId) {
+        User user = userStorage.getUserById(userId);
+        User otherUser = userStorage.getUserById(otherUserId);
+        return user
+                .getFriends()
+                .stream()
+                .filter(friendId -> otherUser.getFriends().contains(friendId))
+                .map(userStorage::getUserById)
+                .collect(Collectors.toList());
+    }
+
+    public List<Film> getFilmsPopular(Long count) {
+        return getFilms()
+                .stream()
+                .filter(film -> film.getLikes() != null)
+                .sorted((t1, t2) -> t2.getLikes().size() - t1.getLikes().size())
+                .limit(count)
+                .collect(Collectors.toList());
+    }
+
+    public void addLike(Long filmId, Long userId) {
+        filmStorage.getFilmById(filmId).getLikes().add(userId);
+    }
+
+    public void deleteLike(Long filmId, Long userId) {
+        userStorage.getUserById(userId);
```

Comment:
Лучше сделать так: 
```suggestion
        User user = userStorage.getUserById(userId);
```
Чтобы не казалось что ты вызываешь этот метод только ради валидации